connectors/http/http-base/src/main/java/io/camunda/connector/http/base/services/CloudFunctionService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.http.base.services;
+
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
+import io.camunda.connector.http.base.auth.BearerAuthentication;
+import io.camunda.connector.http.base.model.ErrorResponse;
+import io.camunda.connector.http.base.model.HttpCommonRequest;
+import io.camunda.connector.http.base.model.HttpMethod;
+import io.camunda.connector.http.base.utils.CloudFunctionHelper;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CloudFunctionService {
```

Comment:
We could have an abstract `ProxyService` and move the GCP-specific implementation into `connectors-bundle-saas`. A couple of things to consider here:
- We might decide to switch to a different proxy in the future - there have been discussions around that
- Customers might want to use something else as a proxy in their own runtime deployments

Reply:
I initially planed to do that. But the authentication here is a bit specific and requires more steps than just using a user/password. It's coming directly from the Google OAuth2 API.

My intention was:
- To keep the current implementation as it worked before, but a bit cleaner.
- Later, add full proxy support (with auth) and see how to integrate our own "proxy" if possible.

It's not a usual proxy per se, it's more that we're forwarding the whole HttpComonRequest to this function. A proxy (in the customer's mind) would be a classic proxy as we know them (haproxy, nginx, etc).

I know it's not ideal, but I couldn't come with a better solution for now at least.
Is that ok or do you still think I should change this code?

Reply:
I agree, makes sense! Let's consider refactoring this separately, if we decide to migrate to a real proxy